### PR TITLE
Simplifying share conversion

### PR DESCRIPTION
**Pull request by benjaminsavage**
Daniel Masny pointed out that the share conversion protocol we had been using was optimized for pushing work into the "offline" phase. In this case, we don't gain any benefit from an "offline" phase, since we are trying to just minimize the total computation cost and latency is not as critical. If we just push it all to the "online" phase, there is actually less communication this way.

### CODE REVIEW

File: src/protocol/modulus_conversion/double_random.rs [hunk 1]
```diff
@@ -106,7 +99,7 @@ impl DoubleRandom {
         record_id: RecordId,
         a: Replicated<F>,
         b: Replicated<F>,
-    ) -> Result<Replicated<F>, Error> {
+    ) -> Result<Replicated<F>, BoxError> {
```
Comment (akoshelev) [inline src/protocol/modulus_conversion/double_random.rs:102]: is something not working with `Error`?

File: src/protocol/modulus_conversion/convert_shares.rs [hunk 2]
```diff
@@ -24,7 +23,6 @@ pub struct ConvertShares {
 #[derive(Debug, Clone, Copy, PartialEq, Eq, Hash)]
 enum Step {
```
Comment (martinthomson) [inline src/protocol/modulus_conversion/convert_shares.rs:24]: Do we still need this?

File: src/protocol/modulus_conversion/double_random.rs (hunk 1 above)
Comment (benjaminsavage) [reply src/protocol/modulus_conversion/double_random.rs:102]: Discussed offline. Seems like we could slowly migrate everything to Error. Maybe not this diff tho.

File: src/protocol/modulus_conversion/convert_shares.rs (hunk 2 above)
Comment (benjaminsavage) [reply src/protocol/modulus_conversion/convert_shares.rs:24]: I kill it in the next diff in this stack: #233

**Review by akoshelev — APPROVED** [review]

**Pull request merged**